dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/UserDetails.java

```diff
@@ -159,6 +162,12 @@ static UserDetails createUserDetails(
               ? setOfIds(user.getTeiSearchOrganisationUnitsWithFallback())
               : (searchOrgUnitUids.isEmpty() ? orgUnitUids : searchOrgUnitUids);
 
+      Set<String> userEffectiveSearchOrgUnitIds =
+          Stream.of(userOrgUnitIds, userSearchOrgUnitIds)
+              .filter(Objects::nonNull)
+              .flatMap(Collection::stream)
```

Comment:
This is a bit odd way to get where `Stream.concat` would get you, like you did in the `User`, but work too I guess :)

Reply:
That's how I originally did it, but sonar complained about a possible NPE, that's why I changed it